cosi/cosi_test.go

```diff
@@ -1,228 +1,92 @@
-	genPostChallengePhaseCosi(cosis, msg)
-	root, children := cosis[0], cosis[1:]
-	var responses []abstract.Scalar
-
-	// for verification later
-	aggResponse := testSuite.Scalar().Zero()
-	for _, ch := range children {
-		// generate the response of each children
-		r, err := ch.CreateResponse()
-		if err != nil {
-			t.Fatal("Error creating response:", err)
-		}
-		responses = append(responses, r)
-		aggResponse = aggResponse.Add(aggResponse, r)
-	}
-	// pass them up to the root
-	_, err := root.Response(responses)
-	if err != nil {
-		t.Fatal("Response phase failed:", err)
-	}
+func TestCoSi(t *testing.T) {
+	n := 5
+	message := []byte("Hello World Cosi")
 
-	// verify it
-	aggResponse = aggResponse.Add(aggResponse, root.response)
-	if !aggResponse.Equal(root.aggregateResponse) {
-		t.Fatal("Responses aggregated not equal")
-	}
-}
-
-// TestCosiSignature test if the signature generation is correct,i.e. if we
-// can verify the final signature.
-func TestCosiSignature(t *testing.T) {
-	msg := []byte("Hello World Cosi")
-	nb := 2
-	cosis := genCosis(nb)
-	genFinalCosi(cosis, msg)
-	root, children := cosis[0], cosis[1:]
+	// Generate key pairs
+	var kps []*config.KeyPair
+	var privates []abstract.Scalar
 	var publics []abstract.Point
-	// add root public key
-	rootPublic := testSuite.Point().Mul(nil, root.private)
-	publics = append(publics, rootPublic)
-	for _, ch := range children {
-		// add children public key
-		public := testSuite.Point().Mul(nil, ch.private)
-		publics = append(publics, public)
-	}
-	sig := root.Signature()
-
-	if err := VerifySignature(testSuite, publics, msg, sig); err != nil {
-		t.Fatal("Error veriying:", err)
-	}
-	var Ed25519Publics []xEd25519.PublicKey
-	for _, p := range publics {
-		buff, err := p.MarshalBinary()
-		assert.Nil(t, err)
-		Ed25519Publics = append(Ed25519Publics, xEd25519.PublicKey(buff))
-	}
-
-	if !cosi.Verify(Ed25519Publics, nil, msg, sig) {
-		t.Error("github.com/bforg/golang-x-crypto/ed25519/cosi fork can't verify")
-	}
-}
-
-func TestCosiSignatureWithMask(t *testing.T) {
-	msg := []byte("Hello World Cosi")
-	nb := 5
-	fail := 2
-	cosis, publics := genCosisFailing(nb, fail)
-	genFinalCosi(cosis, msg)
-	root := cosis[0]
-	sig := root.Signature()
-
-	if err := VerifySignature(testSuite, publics, msg, sig); err != nil {
-		t.Fatal("Error veriying:", err)
+	for i := 0; i < n; i++ {
+		kp := config.NewKeyPair(testSuite)
+		kps = append(kps, kp)
+		privates = append(privates, kp.Secret)
+		publics = append(publics, kp.Public)
 	}
 
-	var Ed25519Publics []xEd25519.PublicKey
-	for _, p := range publics {
-		buff, err := p.MarshalBinary()
-		assert.Nil(t, err)
-		Ed25519Publics = append(Ed25519Publics, xEd25519.PublicKey(buff))
+	// Init masks
+	var masks []*Mask
+	var byteMasks [][]byte
+	for i := 0; i < n; i++ {
+		masks = append(masks, NewMask(testSuite, publics))
+		masks[i].SetMaskBit(i, true) // TODO: this has to go into the mask initialization somehow
+		byteMasks = append(byteMasks, masks[i].mask)
```

Comment:
Mhhh this whole bitmask thing is making me still uneasy ... 
As you state in your TODO, for the moment, the user still have to:
1. Create the mask
2. Set its i bit to true
3. Take its []byte version so it can be used in `AggregateCommitments`

These three steps will **always** be present (you don't create a mask if you don't participante), so it could be good to find a way to make that more transparent, more automatic. Unfortunatly, I don't have a clear idea yet how to do this, I'll think about it...